src/main/java/com/fasterxml/jackson/databind/node/DecimalNode.java

```diff
@@ -132,5 +132,5 @@ public boolean equals(Object o)
     }
 
     @Override
-    public int hashCode() { return Double.valueOf(doubleValue()).hashCode(); }
+    public int hashCode() { return _value.hashCode(); }
```

Comment:
There seemed no reason to use the `hashCode` after converting it to a `Double`, so it was modified to use the `_value` as is.